sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2034,4 +2034,25 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       }
     }
   }
+
+  test("SPARK-21677: json_tuple throws NullPointException when column is null as string type") {
```

Comment:
The end-to-end test at L2047 may not be able to move to `JsonExpressionsSuite`. We can have some unit test cases similar to L2039 in `JsonExpressionsSuite` as @gatorsmile suggested.

It is also good to have similar end-to-end tests in `json-functions.sql`.